### PR TITLE
Use planetscale vitess releases

### PR DESCRIPTION
**Pull request by tomas-didziokas**

### CODE REVIEW

File: libraries/base_service.rb [hunk 1]
```diff
@@ -183,33 +183,62 @@ def bin_location
...
+      def vitess_release
+        node['vitess']['artifacts']['release']
+      end
+
+      def vitess_release_cache_path
         cache_path = Chef::Config[:file_cache_path]
-        bin_name_with_version = "#{version}-#{new_resource.bin_name}"
-        loc = bin_location
+        release_url = node['vitess']['artifacts']['release-url']
+        archyve_file_name = ::File.basename(release_url)
```
Comment (Seitanas) [inline libraries/base_service.rb]: `  archyve_file_name` => `archive_file_name` ?

Comment (tomas-didziokas) [reply libraries/base_service.rb]: 🙇

File: attributes/artifacts.rb [hunk 2]
```diff
@@ -1 +1,7 @@
-default['vitess']['artifacts']['base_url'] = 'https://github.com/vinted/vitess-binary-files/releases/download'
+default['vitess']['artifacts']['release'] = 'b90b3c0'
+
+default['vitess']['artifacts']['release-url'] =
+  'https://github.com/planetscale/vitess-releases/releases/download/b90b3c0/vitess-release-b90b3c0.tar.gz'
```
Comment (ernestas-vinted) [inline attributes/artifacts.rb]: interpolate release-url with release hash

File: attributes/artifacts.rb [hunk 3]
```diff
@@ -1 +1,7 @@
...
+
+default['vitess']['artifacts']['checksum'] =
```
Comment (ernestas-vinted) [inline attributes/artifacts.rb]: move checksum under release since update of release requires new checksum

File: libraries/vtctld_service.rb [hunk 4]
```diff
@@ -60,39 +64,26 @@ def action_delete
...
-          group new_resource.group
-          files_owner new_resource.user
-          files_group new_resource.group
-          files_mode '0640'
-          mode '0755'
-          action :create
+        cache_vitess_binary
+        web_cache_path = ::File.join(vitess_release_cache_path, 'web')
+
+        bash 'copy web assets' do
```
Comment (ernestas-vinted) [inline libraries/vtctld_service.rb]: :clap:

**Comment by ernestas-poskus** [comment]
Also please update every occurence of v3.0.0 with commit hash

```
default['vitess']['version']['mysqlctld'] = 'v3.0.0'
default['vitess']['version']['vtctlclient'] = 'v3.0.0'
default['vitess']['version']['vtctld'] = 'v3.0.0'
default['vitess']['version']['vtgate'] = 'v3.0.0'
default['vitess']['version']['vttablet'] = 'v3.0.0'
default['vitess']['version']['vtworker'] = 'v3.0.0'
default['vitess']['version']['mysqlctl'] = 'v3.0.0'
default['vitess']['version']['vtctl'] = 'v3.0.0'
default['vitess']['version']['vtexplain'] = 'v3.0.0'
default['vitess']['version']['vtbench'] = 'v3.0.0'
```

**Comment by tomas-didziokas** [comment]
@ernestas-poskus i've updated pr, so it will leave ability to install different versions of artifacts

File: attributes/artifacts.rb (hunk 3 above)
Comment (tomas-didziokas) [reply attributes/artifacts.rb]: moved to `default['vitess']['releases']` hash 

```
default['vitess']['releases']['ce90826']['url'] =
  'https://github.com/planetscale/vitess-releases/releases/download/ce90826/vitess-release-ce90826.tar.gz'
default['vitess']['releases']['ce90826']['checksum'] =
  '0af68f031f7840d79c7067973b8234316cb4a8401bb5c9219e602ad693ac886c'
```

**Pull request merged**